pkg/controllers/node/health/controller.go

```diff
@@ -155,11 +164,24 @@ func (c *Controller) findUnhealthyConditions(node *corev1.Node) (nc *corev1.Node
 }
 
 func (c *Controller) annotateTerminationGracePeriod(ctx context.Context, nodeClaim *v1.NodeClaim) error {
+	if expirationTimeString, exists := nodeClaim.ObjectMeta.Annotations[v1.NodeClaimTerminationTimestampAnnotationKey]; exists {
+		expirationTime, err := time.Parse(time.RFC3339, expirationTimeString)
+		if err == nil {
+			if expirationTime.Before(c.clock.Now()) {
+				return nil
+			}
+		}
+	}
```

Comment:
nit can combine them
```suggestion
		if err == nil && expirationTime.Before(c.clock.Now()) {
			return nil
		}
	}
```